Fixup opensearch when using advanced security options

gionn opened the pull request:
Fix #1560

sebastien-rosset commented:
AFAIK, the PR needs to have a changelog fragment. For example: https://github.com/ansible-collections/community.aws/pull/1603/files#diff-91b084c5e7be52b748a76fbb78c48b01265222a931ad815ce7e81a5f0ed17370

markuman commented on plugins/modules/opensearch.py:
```diff
@@ -948,6 +948,7 @@ def set_advanced_security_options(
             ] = advanced_security_opts.get("internal_user_database_enabled")
         master_user_opts = advanced_security_opts.get("master_user_options")
         if master_user_opts is not None:
+            advanced_security_config["MasterUserOptions"] = {}
```
```suggestion
            if not advanced_security_config.get("MasterUserOptions"):
                advanced_security_config["MasterUserOptions"] = {}
```

I'm not sure about it....but I feel better if the empty dict is only set, if `MasterUserOptions` key doesn't exist.  
cc @alinabuzachis

markuman commented:
@gionn yes, as @sebastien-rosset mentioned. Can you please add a changelog fragment?  
And it would be very helpful if you can expand the integration test, that covers this bug.

gionn commented:
I should have addressed all the concerns, but I extended the integration tests blindly and I have no idea if they could work at first try, enabling master user may break the following tests.
